codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java

```diff
@@ -15,60 +15,94 @@
  */
 package io.netty.handler.codec.http;
 
+import io.netty.handler.codec.CharSequenceValueConverter;
 import io.netty.handler.codec.DefaultHeaders;
-import io.netty.handler.codec.DefaultTextHeaders;
-import io.netty.handler.codec.TextHeaders;
+import io.netty.handler.codec.Headers;
+import io.netty.handler.codec.ValueConverter;
 import io.netty.util.AsciiString;
+import io.netty.util.ByteProcessor;
+import io.netty.util.internal.PlatformDependent;
 
+import java.util.AbstractList;
 import java.util.Calendar;
-import java.util.Comparator;
 import java.util.Date;
+import java.util.Iterator;
 import java.util.List;
-import java.util.TreeMap;
+import java.util.Map.Entry;
 
+import static io.netty.util.AsciiString.CASE_INSENSATIVE_HASHER;
+import static io.netty.util.AsciiString.CASE_SENSATIVE_HASHER;
 import static io.netty.util.internal.ObjectUtil.checkNotNull;
 
-public class DefaultHttpHeaders extends DefaultTextHeaders implements HttpHeaders {
-
-    private static final int HIGHEST_INVALID_NAME_CHAR_MASK = ~63;
+public class DefaultHttpHeaders extends DefaultHeaders<CharSequence> implements HttpHeaders {
     private static final int HIGHEST_INVALID_VALUE_CHAR_MASK = ~15;
+    private static final ByteProcessor HEADER_NAME_VALIDATOR = new ByteProcessor() {
+        @Override
+        public boolean process(byte value) throws Exception {
+            switch (value) {
+            case '\t': case '\n': case 0x0b: case '\f': case '\r':
```

Comment:
it's nothing I deeply care about, but this kind of formatting seems to be such an arbitrary thing and makes things hard to read in my opinion.